cpp/include/cudf/utilities/traits.hpp

```diff
@@ -69,6 +81,24 @@ constexpr inline bool is_relationally_comparable()
   return is_relationally_comparable_impl<L, R>::value;
 }
 
+/**
+ * @brief Indicates whether objects of types `L` and `R` can be compared
+ * for equality.
+ *
+ * Given two objects `L l`, and `R r`, returns true if `l == r` and `l > r` is a
```

Comment:
This trait doesn't do any to verify that `l > r` is well formed. 
```suggestion
 * Given two objects `L l`, and `R r`, returns true if `l == r` is a
```